plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/debug/ui/launchConfigurations/EnvironmentTab.java

```diff
@@ -0,0 +1,232 @@
+import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.swt.impl.table.DefaultTableItem;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Represents Environment tab in Launch Configuration
+ * 
+ * @author Oleksii Korniienko olkornii@redhat.com
+ *
+ */
+public class EnvironmentTab extends LaunchConfigurationTab {
+
+	private static final String ADD_SHELL_TITLE = "New Environment Variable";
+	private static final String SELECT_SHELL_TITLE = "Select Environment Variables";
+	private static final String EDIT_SHELL_TITLE = "Edit Environment Variable";
+	private static final String OVERWRITE_SHELL_TITLE = "Overwrite variable?";
+
+	public EnvironmentTab() {
+		super("Environment");
+	}
+
+	/**
+	 * Return the list of all environment variables
+	 *
+	 * @return list of items
+	 */
+	public List<TableItem> getAllVariables() {
+		return new DefaultTable().getItems();
+	}
+
+	/**
+	 * Return the environment variable by name
+	 *
+	 * @return Environment Tab item
+	 */
+	public TableItem getVariable(String variableName) {
+		return new DefaultTable().getItem(variableName);
+	}
+
+	/**
+	 * Return the environment variable by id
+	 *
+	 * @return Environment Tab item
+	 */
+	public TableItem getVariable(int variableId) {
+		return new DefaultTable().getItem(variableId);
+	}
+
+	/**
+	 * Add new environment variable
+	 *
+	 * @param name  variable name
+	 * @param value variable value
+	 */
+	public void add(String name, String value) {
+		new PushButton("Add...").click();
+		new WaitUntil(new ShellIsAvailable(ADD_SHELL_TITLE));
+		new LabeledText("Name:").setText(name);
+		new LabeledText("Value:").setText(value);
+		new OkButton().click();
+		try {
+			new WaitUntil(new ShellIsAvailable(OVERWRITE_SHELL_TITLE), TimePeriod.SHORT);
+			new NoButton().click();
+		} catch (WaitTimeoutExpiredException e) {
+			// variable is new, dont need to overwrite
+		}
+	}
+
+	/**
+	 * Add new environment variable
+	 *
+	 * @param name      variable name
+	 * @param value     variable value
+	 * @param overwrite if variable with the same name exists, then true for
+	 *                  overwrite, else false
+	 */
+	public void add(String name, String value, boolean overwrite) {
+		new PushButton("Add...").click();
+		new WaitUntil(new ShellIsAvailable(ADD_SHELL_TITLE));
+		new LabeledText("Name:").setText(name);
+		new LabeledText("Value:").setText(value);
+		new OkButton().click();
+		try {
+			new WaitUntil(new ShellIsAvailable(OVERWRITE_SHELL_TITLE), TimePeriod.SHORT);
+			if (overwrite) {
+				new YesButton().click();
+			} else {
+				new NoButton().click();
+			}
+		} catch (WaitTimeoutExpiredException e) {
+			// variable is new, dont need to overwrite
+		}
+	}
+
+	/**
+	 * Select environment variable by name
+	 *
+	 * @param variableName variable name for select
+	 */
+	public void selectEnvironmentVariable(String variableName) {
+		new PushButton("Select...").click();
```

Comment:
It would be nice to have this similarities in implementation write only once. Like maybe using functional interface to pass specific code into repeated passages? Ie:
```
	public void versionOne(String param) {
		versionImpl(() -> {
			System.out.println("I am String, joooo!");
		});
	}
	
	public void versionTwo(int param) {
		versionImpl(() -> {
			System.out.println("I am int, jak cyp!!!");
		});
	}
	
	private void versionImpl(Runnable mySpecificImpl) {
		System.out.println("Actions before my functional interface call");
		mySpecificImpl.run();
		System.out.println("Actions after my functional interface call");
	}
```

Reply:
done